service/src/main/java/org/mskcc/cmo/metadb/service/impl/MessageHandlingServiceImpl.java

```diff
@@ -221,13 +224,10 @@ public void run() {
                         MetadbSample existingSample = sampleService.getMetadbSampleByRequestAndIgoId(
                                 sampleMetadata.getRequestId(), sampleMetadata.getPrimaryId());
                         if (existingSample == null) {
-                            LOG.info("Sample metadata does not already exist - persting to db: "
+                            LOG.info("Sample metadata does not already exist - persisting to db: "
                                     + sampleMetadata.getPrimaryId());
                             // handle and persist new sample received
-                            MetadbSample sample = new MetadbSample();
-                            sampleMetadata.setImportDate(
-                                    LocalDateTime.now().format(DateTimeFormatter.ISO_LOCAL_DATE));
-                            sample.addSampleMetadata(sampleMetadata);
+                            MetadbSample sample = SampleDataFactory.buildNewResearchSample(sampleMetadata);
```

Comment:
should we rename this to buildNewResearchSampleFromMetadata ?

Reply:
good idea